web-components/src/components/usa-link/index.js

```diff
@@ -0,0 +1,69 @@
+import { LitElement, css, html } from "lit";
+
+/**
+ * An example element.
+ *
+ * @slot - This element has a slot
+ * @csspart
+ */
+export class UsaLink extends LitElement {
+  static get properties() {
+    return {
+      /**
+       * The destination URL for the link
+       */
+      href: { type: String },
+    };
+  }
+
+  hasLinkChild() {
+    const childLink = this.querySelector("a");
+    if (!childLink) return false;
+
+    this.href = childLink.href;
+    this.slottedChildren = childLink;
+    this.shadowRoot.appendChild(this.slottedChildren);
+  }
+
+  constructor() {
+    super();
+    this.href = "";
+  }
+
+  render() {
+    return this.hasLinkChild()
+      ? html`<a class="usa-link" href="${this.href}"
+          >${this.slottedChildren}</a
+        >`
+      : html`<a class="usa-link" href="${this.href}"><slot></slot></a>`;
+  }
+
+  static styles = css`
```

Comment:
It would be nice to establish a pattern on where styles should live. 

Placing them above `render()` feels familiar to setting a `<link rel="stylesheet">` above HTML markup. Though I can see the benefits of having styles after for convenience while developing.

Reply:
Is there any benefit of having css in JS like this over having a separate `styles.css` sheet and pulling it in via `unsafeCSS`?

I see a benefit of being able to write CSS in a CSS file and keeping `index.js` as JS focused as possible

Reply:
Yeah I don't have any objections to keeping styles in separate files alongside the component code. Definitely helps maintainability. I'll factor them out here.